### PR TITLE
boards: add initial support for stm32g071b-disco board

### PR DESCRIPTION
**Pull request by raiden00pl**
## Summary

- stm32g0: add support for USART3 and USART4
- boards: add initial support for stm32g071b-disco board

## Impact
None
## Testing
nsh

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

File: boards/arm/stm32f0l0g0/stm32g071b-disco/README.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+README
+======
+
+  This README discusses issues unique to NuttX configurations for the
+  STMicro STM32G071B-DISO development board.  The STM32G071B-DISO board
```
Comment (acassis) [inline boards/arm/stm32f0l0g0/stm32g071b-disco/README.txt:5]:
```suggestion
   STMicro STM32G071B-DISCO development board.  The STM32G071B-DISCO board

**Comment by acassis** [comment]
@raiden00pl please add board Documentation to be available at https://nuttx.apache.org/docs/latest/platforms/index.html

File: boards/arm/stm32f0l0g0/stm32g071b-disco/README.txt (hunk 1 above)
Comment (raiden00pl) [reply boards/arm/stm32f0l0g0/stm32g071b-disco/README.txt]: fixed

**Review by hartmannathan — APPROVED** [review]
LGTM

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**